File: trabalho02/FilaCircular/Fila.cs
using System;

class Fila {
    private int capacity = 2;
    private object[] lista = new object[2];
    private int head;
    private int tail;
    private int size;

    public void IncrementSizeList() {
        int newCapacity = capacity * 2;
        object[] novaLista = new object[newCapacity];
        int inicio = head;

        for (int final = 0; final < size; final++) {
            novaLista[final] = lista[inicio];
            inicio = (inicio + 1) % lista.Length;
        }

        tail = size;
        head = 0;
        capacity = newCapacity;
        lista = novaLista;
    }

    public void Enqueue(object v){
        if(IsFull()){
            IncrementSizeList();
        }
        lista[tail] = v;
        tail = (tail + 1) % lista.Length;
        size++;
    }

    public object Dequeue(){
        if(IsEmpty()) throw new EFilaVazia("Fila vazia.");
        object n = lista[head];
        head++;
        return n;
    }

    public bool IsEmpty(){
        return head == tail;
    }

    public bool IsFull(){
        bool isFull = false;
        if(size == capacity) isFull=true;
        return isFull;
    }

    public object First(){
        return lista[head];
    }

    public int Size(){
        return size;
    }

    public void PrintList(){
        for(int i=0; i <= lista.Length - 1; i++){
            Console.Write($"{lista[i]}, ");
        }
        Console.WriteLine();
    }
}


class EFilaVazia : Exception {
 
  public EFilaVazia(string message) : base(message) {
   
  }
  public EFilaVazia(string message, Exception innerException) : base(message, innerException) { }
}

File: edl/unidade1/trabalho03ImplementacoesFila/fila/src/FilaAux/Fila.java
package FilaAux;

public class Fila {
    private int capacity;
    private Object[] lista;
    private int head;
    private int tail;
    private int size;

    public Fila(int c){
        this.capacity = c;
        this.lista = new Object[c];
    }

    public void ed(){
        System.out.print( head );
        System.out.print(" ");
        System.out.print( tail );
        System.out.println();
    }

    public void incrementSizeList() {
        int newCapacity = capacity * 2;
        Object[] novaLista = new Object[newCapacity];
        int inicio = head;

        for (int finalIndex = 0; finalIndex < size; finalIndex++) {
            novaLista[finalIndex] = lista[inicio];
            inicio = (inicio + 1) % lista.length;
        }

        tail = size;
        head = 0;
        capacity = newCapacity;
        lista = novaLista;
    }

    public void enqueue(Object v) {
        if (isFull()) {
            incrementSizeList();
        }
        lista[tail] = v;
        tail = (tail + 1) % lista.length;
        size++;

    }

    public Object dequeue() throws EFilaVazia {
        if (isEmpty()) throw new EFilaVazia("Fila vazia.");
        Object n = lista[head];
        head++;
        size--;

        return n;
    }

    public boolean isEmpty() {
        return head == tail;
    }

    public boolean isFull() {
        return size == capacity;
    }

    public Object first() {
        return lista[head];
    }

    public int size() {
        return size;
    }

    public void printList() {
        for (int i = 0; i < size; i++) {
            System.out.print(lista[i] + ", ");
        }
        System.out.println();
    }
}


